pkg/kubelet/dockershim/network/cni/cni.go

```diff
@@ -376,6 +376,11 @@ func (plugin *cniNetworkPlugin) buildCNIRuntimeConf(podName string, podNs string
 		},
 	}
 
+	//send the annotations as extra args to the cni plugins
+	for key, value := range annotations {
```

Comment:
Especially given the additional context described by @mars1024, I think this functionality is a strong candidate for being extracted to a helper function. We could then easily unit test that helper function.